server/service/src/catalogue/tests/query_catalogue_item.rs

```diff
@@ -0,0 +1,309 @@
+        let context = service_provider.basic_context().unwrap();
+        let service = service_provider.catalogue_service;
+
+        // id equal_any
+        let mut result = service
+            .get_asset_catalogue_items(
+                &context.connection,
+                None,
+                Some(
+                    AssetCatalogueItemFilter::new().id(EqualFilter::equal_any(vec![
+                        "f9e939e2-825d-4b22-a1d0-4983a8c4c2c0".to_owned(),
+                        "c3332de5-a553-4047-a7e4-8ad61325348f".to_owned(),
+                    ])),
+                ),
+                None,
+            )
+            .unwrap();
+
+        assert_eq!(result.count, 2);
+        assert_eq!(result.rows[0].code, "E003/108");
+
+        // id equal_any - no matches
+        result = service
+            .get_asset_catalogue_items(
+                &context.connection,
+                None,
+                Some(
+                    AssetCatalogueItemFilter::new().id(EqualFilter::equal_any(vec![
+                        "id-that-does-not-exist".to_owned(),
+                    ])),
+                ),
+                None,
+            )
+            .unwrap();
+
+        assert_eq!(result.count, 0);
+
+        // category filter - equal to
+        result = service
+            .get_asset_catalogue_items(
+                &context.connection,
+                None,
+                Some(
+                    AssetCatalogueItemFilter::new()
+                        .category(StringFilter::equal_to("Refrigerators and Freezers")),
+                ),
+                None,
+            )
+            .unwrap();
+
+        assert_eq!(result.count, 98);
+
+        // category filter - equal like
+        result = service
+            .get_asset_catalogue_items(
+                &context.connection,
+                None,
+                Some(AssetCatalogueItemFilter::new().category(StringFilter::like("Refrigerato"))),
+                None,
+            )
+            .unwrap();
+
+        assert_eq!(result.count, 98);
+
+        // category filter - no matches
+
+        result = service
+            .get_asset_catalogue_items(
+                &context.connection,
+                None,
+                Some(
+                    AssetCatalogueItemFilter::new()
+                        .category(StringFilter::like("string-with-no-matches")),
+                ),
+                None,
+            )
+            .unwrap();
+
+        assert_eq!(result.count, 0);
+
+        // type filter - equal to
+        result = service
+            .get_asset_catalogue_items(
+                &context.connection,
+                None,
+                Some(
+                    AssetCatalogueItemFilter::new()
+                        .r#type(StringFilter::equal_to("Icelined Refrigerator")),
+                ),
+                None,
+            )
+            .unwrap();
+
+        assert_eq!(result.count, 151);
```

Comment:
```suggestion
        assert_eq!(result.count, 15);
```

This is what I get running the tests, and appears right from looking at the data?